engine/classes/Elgg/Logger.php

```diff
@@ -66,19 +66,37 @@ class Logger {
 	private $printer;
 
 	/**
+	 * @var Config
+	 */
+	private $config;
+
+	/**
 	 * Constructor
 	 *
 	 * @param PluginHooksService $hooks   Hooks service
 	 * @param Context            $context Context service
 	 * @param Printer            $printer Printer
 	 */
-	public function __construct(PluginHooksService $hooks, Context $context, Printer $printer = null) {
+	public function __construct(PluginHooksService $hooks, Context $context, Printer $printer = null, Config $config) {
 		$this->hooks = $hooks;
 		$this->context = $context;
 		if (!isset($printer)) {
 			$printer = new HtmlPrinter();
 		}
 		$this->printer = $printer;
+		$this->config = $config;
+		
+		$php_error_level = error_reporting();
+		
+		if (($php_error_level & E_ALL) == E_ALL) {
+			$this->level = self::INFO;
+		} elseif (($php_error_level & E_NOTICE) == E_NOTICE) {
+			$this->level = self::NOTICE;
+		} elseif (($php_error_level & E_WARNING) == E_WARNING) {
+			$this->level = self::WARNING;
+		} elseif (($php_error_level & E_ERROR) == E_ERROR) {
+			$this->level = self::ERROR;
+		}
```

Comment:
does anyone @Elgg/core  know if this can be done nicer??

Reply:
i think this is beautiful as is :)